### PR TITLE
add graph dumping utilities

### PR DESCRIPTION
**Pull request by yushangdi**
### Description
Add compiler function to dump the forward, backward, and joint graphs. The partitioner is default partition. 
The input meta to each dumped graphs will also be dumped as a pickle file. 

Example usage:

```
    save_fx_func = graph_dumper_aot(current_name, folder_name, dump_example_input = False)
    optimize_ctx = torchdynamo.optimize(
        save_fx_func
    )
    with torch.enable_grad():
        with optimize_ctx:
            result = forward_and_backward_pass(model, example_inputs)
```

### CODE REVIEW

File: functorch/functorch/_src/compilers.py [hunk 1]
```diff
@@ -344,3 +349,116 @@ def debug_compile(fx_g, inps):
...
+        graph_saver_helper(gm, joint_args, "joint")
+        return default_partition(gm, joint_args)
+
+    return aot_module_simplified(gm, fw_compiler=graph_saver_forward,
+                                 bw_compiler=graph_saver_backward,
+                                 partition_fn=graph_saver_joint,
+                                 decompositions=default_decompositions)
+
+
+def get_save_fx_default_func(current_name, folder_name, dump_example_input=False):
```
Comment (Chillee) [inline functorch/functorch/_src/compilers.py:450]: This function name is a bit wordy 🤔 

Could we rename it something like `graph_dumper_aot`

**Review by Chillee — APPROVED** [review]
LGTM

File: functorch/functorch/_src/compilers.py (hunk 1 above)
Comment (yushangdi) [reply functorch/functorch/_src/compilers.py]: sounds good. changed

File: functorch/functorch/_src/compilers.py [hunk 2]
```diff
@@ -344,3 +349,116 @@ def debug_compile(fx_g, inps):
...
+    """
+    Dump the forward, backward, and joint computation graph.
+    Example Usage:
+    save_fx_func = graph_dumper_aot(current_name, folder_name, dump_example_input = False)
+    optimize_ctx = torchdynamo.optimize(
+        save_fx_func
+    )
+    with torch.enable_grad():
+        with optimize_ctx:
+            result = forward_and_backward_pass(model, example_inputs)
```
Comment (IvanYashchuk) [inline functorch/functorch/_src/compilers.py:460]: Where is `forward_and_backward_pass` defined?
